rfcs/2023-11-08-font-loading-strategy.md

```diff
@@ -0,0 +1,211 @@
+  font-style: normal;
+}
+```
+
+CSR apps can also add preloads to fonts to load them faster
+
+```html
+<link rel="preload" href="/fonts/TASAOrbiterVF.woff2" as="font" type="font/woff2" crossorigin />
+<link rel="preload" href="/fonts/Inter-Regular.ttf" as="font" type="font/ttf" crossorigin />
+<link rel="preload" href="/fonts/Inter-SemiBold.ttf" as="font" type="font/ttf" crossorigin />
+```
+
+Example: https://stackblitz.com/~/github.com/saurabhdaware/blade-self-hosted-font
+
+</details>
+
+<!-- #### Difference between loading fonts without preload and with preload
+
+https://github.com/razorpay/blade/assets/30949385/b2b12cbe-2c93-4af9-bc7f-7b94ed36b90b
+
+https://github.com/razorpay/blade/assets/30949385/d468cfc1-c1a9-44ad-9d85-a3ea37320ce7 -->
+
+#### 2. Self Hosted: Using CSS Imports from Blade (✅ Preferred)
+
+Similar to how we load Lato font from `@fontsource/lato` right now, Consumers add import from `@razorpay/blade/fonts.css`
+
+```js
+import '@razorpay/blade/fonts.css';
+```
+
+**Pros:**
+
+- Easy Implementation
+- Self-hosted so does not require additional DNS connection and thus fonts load faster
+- Font names, font files, fallbacks for browsers, definition of fonts, is taken care by Blade
+
+**Cons:**
+
+- Fonts cannot be preloaded with HTML preload in CSR apps (SSR apps can loop through their build manifest to know the font URLs and preload them if required)
+
+#### 3. Host fonts on Razorpay's CDN
+
+Similar to how you install from google font, except we can have `https://cdn.razorpay.com/blade/fonts.css` kind of URL.
+
+**Pros:**
+
+- Easy Implementation
+- Font definitions are taken care by blade
+- Can be preloaded
+
+**Cons:**
+
+- More implementation and maintainance cost since we don't have well-owned CDN currently
+- Requires additional DNS connection before fonts start to load
+- Unlike method 2, this cannot be versioned or easily be kept in-sync with Blade's version
+
+#### 4. Load from existing third-party CDN
+
+While Inter can be loaded from [Google Fonts](https://fonts.google.com/specimen/Inter), Tasa Orbiter can be loaded from [CDNFonts](https://www.cdnfonts.com/tasa-orbiter-display.font)
+
+Cons from all of the above 🙈
+
+## Implementation Details
+
+### Variable Fonts vs Static Fonts
+
+Should Blade load variable fonts or static-fonts?
+
+The performance of variable font vs static font depends on how many weights and styles are being used, the size of that particular font in itself, etc.
+
+These are the weights and styles being used right now
+
+**Tasa Orbiter:** 400, 500, 600
+**Inter:** 400, 600
+
+**Conclusion**
+
+- We use Tasa Orbiter as variable font since its 32kb in size compared to ~90kb (30kb x 3 weights) of static font
+- We use Inter as static font. We use lighter version of Inter from [@fontsource/inter](https://www.npmjs.com/package/@fontsource/inter?activeTab=code) as static fonts there are 70kb each compared to ~200kb on Google Fonts version. The variable font of Inter from Google Fonts is ~800kb.
```

Comment:
can't we use variable fonts for inter too as for Tasa and store it locally like we do for tasa?

Reply:
we can. Its just heavier. Its 200kb vs ~140kb for addition of our required weights. So 60kb extra 🤔

Reply:
Added a table here to represent information better